src/core/json-schema/src/formly-json-schema.service.ts

```diff
@@ -70,6 +80,18 @@ export class FormlyJsonschema {
       }
     }
 
+    // map in possible formlyConfig options from the widget property
+    if (jsonSchema['widget'] && jsonSchema['widget'].formlyConfig) {
```

Comment:
I'm asking myself if `widget` is a valid `json-schema` or not :thinking:

Reply:
http://json-schema.org/draft-07/schema

This states you can have any additionalProperties added to a valid jsonSchema json file.  This is basically a jsonSchema that defines jsonSchema syntax :0.  Any who, because of that definition, it is valid to have any fields we want to add to it on it.  The caveat being that those fields won't be parsed by other validators / ui engines unless they explicitly add it.  This will be remedied in draft-08 where we are getting the ability to add "vocabularies" to schemas. That allows implementers to create extensions based on vocabularies and share the vocabulary (e.g. the react json schema guys are working on the UISchema volcabulary).

When all said and done, I do want to make a hyper schema that uses the widget volcabulary so we can have people make valid layout.json with this library without having to go between IDE and browser checking for syntax errors.

I also want to make a companion app that will allow drag/drop layout fields based on the dataSchema for more non technical users to be able to create layout.json but that has nothing to do with your comment haha.

Reply:
Yes me too. I've seen it in various places being used also by others (as mentioned in our discussion issue). But it seemed strange as it wasn't included as a valid property in the typings file..

Reply:
Oh, this is also why I wanted to make sure we had the split concept of "dataSchema" vs "layoutSchema". the data schema should define available fields while the layoutSchema defines what the layout should look like.  We can merge and things behind the scenes still for ease of processing of the fields.  We can also define a  "combinedSchema" concept that still allows for users to just provide that alone.

Reply:
> This states you can have any additionalProperties added to a valid jsonSchema json file. This is basically a jsonSchema that defines jsonSchema syntax :0. Any who, because of that definition, it is valid to have any fields we want to add to it on it.

@kenisteward if I got what you mean 😅  then that means it's fine to add the `widget` (which also confirms why I've seen it in so many other places

@kenisteward I don't quite get your last comment though. I mean, sure, I always agree to have separation of concerns, thus having two different schemas, for data & UI. That would be the JSONSchema and the UISchema, right?
I just thought we agreed to go with the `widget` so far that somehow is the "layoutSchema" integrated into the JSONSchema for now: https://github.com/ngx-formly/ngx-formly/issues/1459#issuecomment-470631065

That way it's easier to use for ppl 😄. Actually I'm using this PR in my project for experimentation purposes and it's really easy that way to use JSONSchema to create forms. Also the impact on ngx-formly is rather small for now

Reply:
in the end, we'll provide both approaches, IMO the `map` callback solved almost everything, 
 WDYT to consider the `widget` as an addon which can be imported by end users in order to use it.
```ts
import { FormlyJsonschema, widgetAddon  } from '@ngx-formly/core/json-schema';

...
fields: FormlyFieldConfig[] = [
  this.formlyJsonschema.toFieldConfig(
    jsonSchema,
    { map: widgetAddon }
  )
]
```

Reply:
well just leave as it, the current approach seems good anyway.

Reply:
@juristr 
>@kenisteward if I got what you mean 😅 then that means it's fine to add the widget (which also confirms why I've seen it in so many other places

Yup, that's right!

>I just thought we agreed to go with the widget so far that somehow is the "layoutSchema" integrated into the JSONSchema for now: #1459 (comment)

Doing it this way doesn't allow someone who wants to separate their dataSchema from their uiSchema and share the dataSchema with the backend for validation.  Technically, syntactically you can do this and the backend parser will ignore all widget fields but architecturally if one wanted to have something that just represented the data then they couldn't do that.  They would have to write a mapper that has keyValue pairs of property key to layout field when all they would need is a configuration.

To refer to the example @aitboudad made, I would consider that a JSONSchema that represents how the ui renderings not a JSONSchema that represents how the data should be submitted.  Because of some specifics of how it gets translated in ngx-formly, it did fit the example of the model data that got spit out but there was no representation of the base schema I gave.

@aitboudad 
> WDYT to consider the `widget` as an addon which can be imported by end users in order to use it.

This makes sense if we want to say only support either just the dataSchema (default way we have it now) or dataSchema merged with widget (which you have it for your addon) but I don't think it would be for I have a JSONSchema that represents the data and I have a separate JSONSchema that represents the UI Layout.  Or would you want to do something like have a  "dualSchemaAddon"  that allows you to pass 2 different things to represent each part?